sklearn/tests/test_calibration.py

```diff
@@ -0,0 +1,180 @@
+                      cluster_std=3.0)
+    X_train, y_train = X[::2], y[::2]
+    X_test, y_test = X[1::2], y[1::2]
+    for method in ['isotonic', 'sigmoid']:
+        cal_clf = CalibratedClassifierCV(clf, method=method, cv=2)
+        cal_clf.fit(X_train, y_train)
+        probas = cal_clf.predict_proba(X_test)
+        assert_array_almost_equal(np.sum(probas, axis=1), np.ones(len(X_test)))
+        assert_true(clf.fit(X_train, y_train).score(X_test, y_test) <=
+                    cal_clf.fit(X_train, y_train).score(X_test, y_test))
+
+    # Test that calibration of a multiclass classifier decreases log-loss
+    X, y = make_blobs(n_samples=100, n_features=2, random_state=42,
+                      cluster_std=3.0)
+    X_train, y_train = X[::2], y[::2]
+    X_test, y_test = X[1::2], y[1::2]
+
+    clf = RandomForestClassifier(n_estimators=10, random_state=42)
+    clf.fit(X_train, y_train)
+    clf_probs = clf.predict_proba(X_test)
+    score = log_loss(y_test, clf_probs)
+
+    for method in ['isotonic', 'sigmoid']:
+        cal_clf = CalibratedClassifierCV(clf, method=method, cv=3)
+        cal_clf.fit(X_train, y_train)
+        cal_clf_probs = cal_clf.predict_proba(X_test)
+        cal_score = log_loss(y_test, cal_clf_probs)
+        assert_true(cal_score < score)
+
+
+def test_calibration_prefit():
+    """Test calibration for prefitted classifiers"""
+    n_samples = 50
+    X, y = make_classification(n_samples=3 * n_samples, n_features=6,
+                               random_state=42)
+    sample_weight = np.random.RandomState(seed=42).uniform(size=y.size)
+
+    X -= X.min()  # MultinomialNB only allows positive X
+
+    # split train and test
+    X_train, y_train, sw_train = \
+        X[:n_samples], y[:n_samples], sample_weight[:n_samples]
+    X_calib, y_calib, sw_calib = \
+        X[n_samples:2*n_samples], y[n_samples:2*n_samples], \
+        sample_weight[n_samples:2*n_samples]
+    X_test, y_test, sw_test = \
+        X[2*n_samples:], y[2*n_samples:], sample_weight[2*n_samples:]
+
+    # Naive-Bayes
+    clf = MultinomialNB()
+    clf.fit(X_train, y_train, sw_train)
+    prob_pos_clf = clf.predict_proba(X_test)[:, 1]
+
+    # Naive Bayes with calibration
+    for this_X_calib, this_X_test in [(X_calib, X_test),
+                                      (sparse.csr_matrix(X_calib),
+                                       sparse.csr_matrix(X_test))]:
+        for method in ['isotonic', 'sigmoid']:
+            pc_clf = CalibratedClassifierCV(clf, method=method, cv="prefit")
+            pc_clf.fit(this_X_calib, y_calib, sample_weight=sw_calib)
+            prob_pos_pc_clf = pc_clf.predict_proba(this_X_test)[:, 1]
+
+            assert_true(brier_score_loss(y_test, prob_pos_clf) >
+                        brier_score_loss(y_test, prob_pos_pc_clf))
+
+
+def test_sigmoid_calibration():
+    """Test calibration values with Platt sigmoid model"""
+    exF = np.array([5, -4, 1.0])
+    exY = np.array([1, -1, -1])
```

Comment:
No idea, this test was probably written by @agramfort before I joined working on this PR